Stress analyzer/ViewModels/AnalyticsViewModel.swift

```diff
@@ -13,25 +13,32 @@ final class AnalyticsViewModel: ObservableObject {
     /// of surveyID, and values of [ChartData] which
     /// represent the list of ChartData objects for
     /// each question of the survey.
-    @Published var ResponseRatesBySurvey: [Int: [ChartData]]
+    @Published var ResponseRatesBySurvey: [ChartData]
     @Published var personalityScores: [String: PersonalityScore]
     let baseURL = "http://swiftware.tech"
 
     init() {
         // New class properties should be initialized in here
-        self.ResponseRatesBySurvey = [Int : [ChartData]]()
+        self.ResponseRatesBySurvey =  [ChartData]()
         self.personalityScores = [String: PersonalityScore]()
     }
 
-    // TODO: Implement Function
-    /// Returned Data should be saved into ResponseRatesBySurvey
-    /// Dictionary. Responses from the server come back as a list
-    /// of ChartData objects, one ChartData instance per question
-    ///  on the server. So if the returned ChartData items from
-    ///  the server for surveyID: 1 is `chartDataList`, then it
-    ///  can be saved using `ResposneRatesBySurvey[1] = chartDataList`
     func getAvgResponseRate(surveyID : Int) async {
-
+        let url = URL(string: baseURL + "avgResponseRate/" + String(surveyID))!
+        //print ("url: " , url)
+        var request = URLRequest(url: url)
+        request.httpMethod = "GET"
+        
+        do {
+            print ("inside do task")
+            let (data, _) = try await URLSession.shared.data(for: request)
+            ResponseRatesBySurvey = try JSONDecoder().decode([ChartData].self, from: data)
```

Comment:
Hey @adesh405, for the purpose of this assignment this looks good, but just so you know, the reason I initially used a map of [int: [ChartData]] was so that we could store the avgResponseRates for all surveys instead of just one. I don't realistically see us changing this by the end of the project, but if we do, then it should be an easy enough change to replace this line with

`ResponseRatesBySurvey[surveyID] = try JSONDecoder().decode([ChartData].self, from: data)`